src/threadlist.cpp

```diff
@@ -686,19 +687,59 @@ ThreadList::waitForAllRestored(Thread *thread)
     sem_wait(&semWaitForCkptThreadSignal);
   }
   Thread_RestoreSigState(thread);
+
+  if (thread == motherofall) {
+    /* If DMTCP_RESTART_PAUSE==4, sleep 15 seconds to allow gdb attach.*/
+    char * pause_param = getenv("DMTCP_RESTART_PAUSE");
+    if (pause_param == NULL) {
+      pause_param = getenv("MTCP_RESTART_PAUSE");
+    }
+    if (pause_param != NULL && pause_param[0] == '4' &&
+        pause_param[1] == '\0') {
+#ifdef HAS_PR_SET_PTRACER
+      prctl(PR_SET_PTRACER, PR_SET_PTRACER_ANY, 0, 0, 0); // For: gdb attach
+#endif // ifdef HAS_PR_SET_PTRACER
+      struct timespec delay = { 15, 0 }; /* 15 seconds */
+      printf("Pausing 15 seconds. Do:  gdb <PROGNAME> %d\n",
+             dmtcp_virtual_to_real_pid(getpid()));
+      nanosleep(&delay, NULL);
+#ifdef HAS_PR_SET_PTRACER
+      prctl(PR_SET_PTRACER, 0, 0, 0, 0); // Revert permission to default.
+#endif // ifdef HAS_PR_SET_PTRACER
+    }
+  }
 }
 
 /*****************************************************************************
  *
  *****************************************************************************/
 void
+ThreadList::postRestartDebug(double readTime)
+{ // Don't try to print before debugging.  Who knows what is working yet?
+  int dummy = 1;
+#ifndef DEBUG
+  // printf may fail, but we'll risk it to let user know this:
+  printf("\n** DMTCP: It appears DMTCP not configured with '--enable-debug'\n");
+  printf("**        If GDB doesn't show source, re-configure and re-compile\n");
+#endif
+  while (dummy);
+  // User should have done GDB attach if we're here.
+  prctl(PR_SET_PTRACER, 0, 0, 0, 0); // Revert permission to default: no ptracer
```

Comment:
Sure, but we still need to wrap the call to prctl in `#ifdef HAS_PR_SET_PTRACER`.